### ISSUE TITLE
Fix issues with ip permisions for egress rules - similar to Issue 2775 for ingress

### ISSUE THREAD

**Issue by jmbollard** [spulec/moto#3245]
The first issue is adding the ability to add a Description to the IpPermissions of an egress rule, which was addressed for ingress rules in [PR 3196](https://github.com/spulec/moto/pull/3196). 

Using version 1.3.15.dev1023 of moto, the following test for an ip permissions on an egress rule with a description fails:

```
import boto3
from moto import mock_ec2

some_ip = "54.1.1.2"
some_instance_id = "i-0db65f4f36b4733aa"
user_id_group_pair = {"UserIdGroupPairs": []}
default_egress_ip_perms = {
    "IpProtocol": "-1",
    "IpRanges": [{"CidrIp": "0.0.0.0/0"}],
    "UserIdGroupPairs": [],
}

@mock_ec2
def test_sg_ip_range_with_description():
    ec2_resource = boto3.resource("ec2", "us-east-1")
    some_public_sg = ec2_resource.create_security_group(
        Description="Some public security group",
        GroupName="SomeGroupName",
        VpcId="SomeVpcId",
    )
    some_ip_perm_with_desc = {
        "IpProtocol": "50",
        "IpRanges": [{"CidrIp": some_ip + "/32", "Description": some_instance_id}],
    }
    some_ip_perm_with_desc.update(user_id_group_pair)`

    some_public_sg.authorize_egress(IpPermissions=[some_ip_perm_with_desc])

    # This test fails
    assert some_ip_perm_with_desc in some_public_sg.ip_permissions_egress
    # This test passes if run alone
    assert default_egress_ip_perms in some_public_sg.ip_permissions_egress
```

 The same change that was made for ingress can be made for the Egress rules in moto/ec2/responses/security_groups.py in the <ipPermissionsEgress> element.  This fixes the issue with the Description, but breaks the test for the default egress rule because the default rule is created with an ip range of ["0.0.0.0/0"] instead of [{'CidrIp": "0.0.0.0"}], which causes security_group.ip_permissions_egress to return an ip permissions with no CidrIp value i.e.  [{'CidrIp': ''}]

I have made additional changes to address this issue and will be submitting a PR for review shortly with what I believe are the necessary changes.

**Comment by bblommers** [comment]
Thanks for raising this, @jmbollard, PR's are always welcome!

**Comment by bblommers** [comment]
This was fixed with #3250
